### PR TITLE
Update script-loader.php

### PR DESCRIPTION
**Pull request by kbiegert**
Prevent null pointer.

Trac ticket: [ticket 56482](https://core.trac.wordpress.org/ticket/56482#ticket)

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/script-loader.php
Comment (costdev) [inline src/wp-includes/script-loader.php]:
```suggestion
		get_current_screen() !== null &&
```
While this is comparing to `null`, it will still generate a warning in PHPCS (not a failure though). Despite this, to keep noise down in logs, use the strictest comparison possible.

**Review by costdev — CHANGES_REQUESTED** [review]
Thanks for the PR @kbiegert! I've left one small suggestion.

Can you also update the PR's description to include a link to the Trac ticket for this patch? Thanks!

File: src/wp-includes/script-loader.php
Comment (costdev) [inline src/wp-includes/script-loader.php:2408]: Rather than call `get_current_screen()` twice, can we add `global $current_screen;` at the top of the function (above the multiline comment, with a line break), and have both of these conditions use the `$current_screen` variable instead? The docblock for this function will also need to have a `@global` tag added:

```
* @since 5.9.1
*
* @global WP_Screen $current_screen WordPress current screen object.
*/
```

Also, can we replace the new condition with `$current_screen instanceof WP_Screen`? This will ensure that it's a `WP_Screen` object before attempting to call the `::is_block_editor()` method in the last condition.

**Review by costdev — CHANGES_REQUESTED** [review]
Thanks for the quick update @kbiegert! Just a couple more suggestions 🙂

File: src/wp-includes/script-loader.php
Comment (costdev) [reply src/wp-includes/script-loader.php:2408]: Actually, hold off on the above suggestion for now. Pinging @SergeyBiryukov for a second opinion on this.

Sergey, some files in Core have multiple calls to `get_current_screen()`. Normally, we'd avoid multiple function calls. Is this something we should avoid here, or is this an exception?

Comment (jrfnl) [reply src/wp-includes/script-loader.php:2408]: Looking at the contents of the [`get_current_screen()`](https://developer.wordpress.org/reference/functions/get_current_screen/) function, @costdev's suggestions make sense to me.

I'd also would like to see a test covering the originally reported issue (throwing an error on PHP 8.1 without the fix and passing with the fix).

Comment (costdev) [reply src/wp-includes/script-loader.php:2408]: Regarding the test: The function relies on `is_admin()`, which, based on [its source](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/load.php#L1173) will require `WP_ADMIN` to be defined in order to test this issue and fix.

`WP_ADMIN` isn't defined in the test suite, and doing so in this test would impact other tests (including for this function). Not sure how to proceed with that. 

That said, here's two screenshots:

As-is:
image

With new condition:
image

Comment (jrfnl) [reply src/wp-includes/script-loader.php:2408]: Yes, you got a point about the `define()`. There are ways around it (run in separate process, run as separate test suite), but those are outside the scope of this particular fix.

**Pull request closed without merging**
